src/containers/PageBuilder/Primitives/SearchCTA/SearchCTA.js

```diff
@@ -0,0 +1,176 @@
+import FilterKeyword from './FilterKeyword/FilterKeyword';
+
+import css from './SearchCTA.module.css';
+
+const GRID_CONFIG = [
+  { gridCss: css.gridCol1 },
+  { gridCss: css.gridCol2 },
+  { gridCss: css.gridCol3 },
+  { gridCss: css.gridCol4 },
+];
+
+const getGridCount = numberOfFields => {
+  const gridConfig = GRID_CONFIG[numberOfFields - 1];
+  return gridConfig ? gridConfig.gridCss : GRID_CONFIG[0].gridCss;
+};
+
+const isEmpty = value => {
+  if (value == null) return true;
+  return value.hasOwnProperty('length') && value.length === 0;
+};
+
+const formatDateValue = (dateRange, queryParamName) => {
+  const hasDates = dateRange;
+  const { startDate, endDate } = hasDates ? dateRange : {};
+  const start = startDate ? stringifyDateToISO8601(startDate) : null;
+  const end = endDate ? stringifyDateToISO8601(endDate) : null;
+  const value = start && end ? `${start},${end}` : null;
+  return { [queryParamName]: value };
+};
+
+export const SearchCTA = React.forwardRef((props, ref) => {
+  const history = useHistory();
+  const routeConfiguration = useRouteConfiguration();
+  const config = useConfiguration();
+
+  const { categories, dateRange, keywordSearch, locationSearch } = props.searchFields;
+
+  const [submitDisabled, setSubmitDisabled] = useState(false);
+
+  const categoryConfig = config.categoryConfiguration;
+
+  const filters = {
+    categories: {
+      enabled: categories,
+      isValid: () => categoryConfig.categories.length > 0,
+      render: () => (
+        <div className={css.filterField}>
+          <FilterCategories categories={categoryConfig.categories} />
+        </div>
+      ),
+    },
+    locationSearch: {
+      enabled: locationSearch,
+      isValid: () => locationSearch,
+      render: () => (
+        <div className={css.filterField}>
+          <FilterLocation setSubmitDisabled={setSubmitDisabled} />
+        </div>
+      ),
+    },
+    keywordSearch: {
+      enabled: keywordSearch,
+      isValid: () => keywordSearch,
+      render: () => (
+        <div className={css.filterField}>
+          <FilterKeyword />
+        </div>
+      ),
+    },
+    dateRange: {
+      enabled: dateRange,
+      isValid: () => dateRange,
+      render: () => (
+        <div className={css.filterField}>
+          <FilterDateRange config={config} />
+        </div>
+      ),
+    },
+  };
+
+  const addFilterMaybe = key => {
+    const filter = filters[key];
+    return filter.enabled && filter.isValid() ? filter.render() : null;
+  };
+
+  // Count the number search fields that are enabled
+  const fieldCountForGrid = Object.values(filters).filter(field => field.enabled && field.isValid())
+    .length;
+
+  //  If no search fields are enabled, we return null (Console won't allow you to enable 0 search fields)
+  if (!fieldCountForGrid) {
+    return null;
+  }
+
+  const onSubmit = values => {
+    // Convert form values to query parameters
+    const queryParams = {};
```

Comment:
```suggestion
    let queryParams = {};
```
This is not a constant, so [let](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let) is more appropriate.

JavaScript's **_const_** does not block a re-assignment to nested properties of an object, but it's still good to be transparent what's the use case.